mailbox/event/event-memory/src/test/java/org/apache/james/mailbox/events/delivery/InVmEventDeliveryTest.java

```diff
@@ -53,6 +57,77 @@ void setUp() {
         inVmEventDelivery = new InVmEventDelivery(new NoopMetricFactory());
     }
 
+    @Nested
+    class ErrorHandling {
+
+        @Test
+        void deliverShouldReturnSuccessResultWhenSynchronousListenerFailsLEMaxRetry() {
+            when(listener.getExecutionMode()).thenReturn(MailboxListener.ExecutionMode.SYNCHRONOUS);
+
+            doThrow(RuntimeException.class)
+                .doThrow(RuntimeException.class)
+                .doThrow(RuntimeException.class)
+                .doNothing()
+                .when(listener).event(event);
+
+            assertThatCode(() -> inVmEventDelivery.deliver(ImmutableList.of(listener), event)
+                    .allListenerFuture()
+                    .block())
+                .doesNotThrowAnyException();
+        }
+
+
+        @Test
+        void deliverShouldReturnErrorResultWhenSynchronousListenerFailsGTMaxRetry() {
+            when(listener.getExecutionMode()).thenReturn(MailboxListener.ExecutionMode.SYNCHRONOUS);
+
+            doThrow(RuntimeException.class)
+                .doThrow(RuntimeException.class)
+                .doThrow(RuntimeException.class)
+                .doThrow(RuntimeException.class)
+                .doNothing()
+                .when(listener).event(event);
+
+            assertThatThrownBy(() -> inVmEventDelivery.deliver(ImmutableList.of(listener), event)
+                    .allListenerFuture()
+                    .block())
+                .isInstanceOf(RuntimeException.class);
+        }
+
+        @Test
+        void deliverShouldReturnSuccessResultWhenAsynchronousListenerFailsLEMaxRetry() {
+            when(listener2.getExecutionMode()).thenReturn(MailboxListener.ExecutionMode.ASYNCHRONOUS);
+
+            doThrow(RuntimeException.class)
+                .doThrow(RuntimeException.class)
+                .doThrow(RuntimeException.class)
+                .doNothing()
+                .when(listener2).event(event);
+
+            assertThatCode(() -> inVmEventDelivery.deliver(ImmutableList.of(listener2), event)
+                    .allListenerFuture()
+                    .block())
+                .doesNotThrowAnyException();
+        }
+
+        @Test
+        void deliverShouldReturnErrorResultWhenAsynchronousListenerFailsGTMaxRetry() {
+            when(listener2.getExecutionMode()).thenReturn(MailboxListener.ExecutionMode.ASYNCHRONOUS);
+
+            doThrow(RuntimeException.class)
+                .doThrow(RuntimeException.class)
+                .doThrow(RuntimeException.class)
+                .doThrow(RuntimeException.class)
+                .doNothing()
+                .when(listener2).event(event);
+
+            assertThatThrownBy(() -> inVmEventDelivery.deliver(ImmutableList.of(listener2), event)
+                    .allListenerFuture()
+                    .block())
+                .isInstanceOf(RuntimeException.class);
```

Comment:
IMO you are not supposed to fail here

Reply:
See https://github.com/linagora/james-project/pull/2086#discussion_r246277374 thread